### PR TITLE
Directory Structure Implementation

### PR DESCRIPTION
**Pull request by JeyJeyGao**
1. support dir package
2. supoort notation-go/config package

Test:
tested with the workflow mentioned in [hello-signing](https://github.com/notaryproject/notation/blob/main/docs/hello-signing.md)
1. `notation cert generate-test --default "wabbit-networks.io"`
2. `notation --plain-http sign $IMAGE`
3. `notation --plain-http list $IMAGE`
4. `notation --plain-http verify $IMAGE`
5. `notation cache list`
6. `notation plugin list`
7. `notation pull $IMAGE`
8. `notation push $IMAGE`
9. `notation key list`
10. `notation cert add --name "wabbit-networks.io" ~/.config/notation/localkeys/wabbit-networks.io.crt`

Signed-off-by: Junjie Gao <43160897+JeyJeyGao@users.noreply.github.com>

### CODE REVIEW

File: cmd/notation/cert_gen.go [hunk 1]
```diff
@@ -45,21 +46,21 @@ func generateTestCert(ctx *cli.Context) error {
 	fmt.Println("generated certificates expiring on", cert.NotAfter.Format(time.RFC3339))
 
 	// write private key
-	keyPath := config.KeyPath(name)
+	keyPath := dir.Path.Localkey(name + dir.KeyExtension)
```
Comment (binbin-li) [inline cmd/notation/cert_gen.go:49]: I feel passing in 2 vars: `name` and `extention` might be more clear than constructing a name with extention manually.

Comment (JeyJeyGao) [reply cmd/notation/cert_gen.go]: Done. Thanks!

File: pkg/auth/native_store.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ func GetCredentialsStore(registryHostname string) (CredentialStore, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to load config file, error: %v", err)
 	}
+	fmt.Printf("test: configFile: %+v, %s", configFile, registryHostname)
```
Comment (patrickzheng200) [inline pkg/auth/native_store.go:33]: Looks like this is a test log? We should remove it.

File: pkg/auth/native_store.go [hunk 3]
```diff
@@ -49,7 +50,8 @@ func newNativeAuthStore(helperSuffix string) CredentialStore {
 // getConfiguredCredentialStore returns the credential helper configured for the
 // given registry, the default credsStore, or the empty string if neither are
 // configured.
-func getConfiguredCredentialStore(c *config.File, registryHostname string) string {
+func getConfiguredCredentialStore(c *config.Configuration, registryHostname string) string {
+	fmt.Printf("CredentialHelpers %v", c.CredentialHelpers)
```
Comment (patrickzheng200) [inline pkg/auth/native_store.go:54]: Same here?

File: pkg/auth/native_store.go (hunk 2 above)
Comment (JeyJeyGao) [reply pkg/auth/native_store.go]: Deleted.

File: pkg/auth/native_store.go (hunk 3 above)
Comment (JeyJeyGao) [reply pkg/auth/native_store.go]: Deleted.

**Review by binbin-li — APPROVED** [review]
lgtm

**Review by chloeyin — APPROVED** [review]
lgtm

**Review by patrickzheng200 — APPROVED** [review]
lgtm

**Comment by dtzar** [comment]
Per the [PR review criteria](https://github.com/notaryproject/notation/discussions/263) @notaryproject/notation-sub-group-maintainers - this is past a week old and has two approvals from official code-reviewers, so it is eligible for merge now.

File: pkg/config/once.go [hunk 4]
```diff
@@ -1,20 +1,59 @@
...
+type KeySuite = config.KeySuite
+
+// ExternalKey contains the necessary information to delegate the signing operation to the named plugin.
+type ExternalKey = config.ExternalKey
+
+// X509KeyPair contains the paths of a public/private key pair files.
+type X509KeyPair = config.X509KeyPair
+
+//CertificateReference is a named file path.
+type CertificateReference = config.CertificateReference
```
Comment (shizhMSFT) [inline pkg/config/once.go:35]: Curious why we don't call `github.com/notaryproject/notation-go/config` directly and rename this package to something like `configutil`?

Comment (JeyJeyGao) [reply pkg/config/once.go]: Thanks for your suggestion.

File: cmd/notation/cert.go [hunk 5]
```diff
@@ -8,6 +8,7 @@ import (
 	"time"
 
 	"github.com/notaryproject/notation-core-go/x509"
+	configutil "github.com/notaryproject/notation-go/config"
```
Comment (shizhMSFT) [inline cmd/notation/cert.go:11]: Let's keep `github.com/notaryproject/notation-go/config` as `config`, and rename `github.com/notaryproject/notation/pkg/config` as `github.com/notaryproject/notation/pkg/configutil`.

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: cmd/notation/cert.go (hunk 5 above)
Comment (JeyJeyGao) [reply cmd/notation/cert.go]: Done and tested again.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**